jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/DMatrix.java

```diff
@@ -395,6 +395,39 @@ public float[] getBaseMargin() throws XGBoostError {
     return getFloatInfo("base_margin");
   }
 
+  /**
+   * Get feature data as BigDenseMatrix
+   * @return feature Matrix
+   * @throws XGBoostError
+   */
+  public BigDenseMatrix getData() throws XGBoostError {
+    int rowNum = (int) rowNum();
+    int colNum = (int) colNum();
+    int nonMissingNum = (int) nonMissingNum();
+
+    long[] rowOffset = new long[rowNum + 1];
+    int[] featureIndex = new int[nonMissingNum];
+    float[] featureValue = new float[nonMissingNum];
+
+    XGBoostJNI.checkCall(XGBoostJNI.XGDMatrixGetDataAsCSR(handle, "{}",
```

Comment:
@trivialfis, is there an API to get the dense matrix directly instead of converting CSR data to the dense dmatrix?

Reply:
After checking the source code, I didn't find such functions. But @trivialfis can confirm it.

Reply:
No, DMatrix itself is a CSR matrix.